subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/RichVersionConstraintsIntegrationTest.groovy

```diff
@@ -396,19 +397,23 @@ class RichVersionConstraintsIntegrationTest extends AbstractModuleDependencyReso
 
         resolve.expectGraph {
             root(":", ":test:") {
-                edge("org:foo:{strictly $directDependencyVersion}", "org:foo:1.2")
+                edge("org:foo:{strictly $directDependencyVersion}", "org:foo:$resolvedVersion")
                 edge("org:bar:1.0", "org:bar:1.0") {
-                    edge("org:foo:$transitiveDependencyVersion", "org:foo:1.2")
+                    edge("org:foo:$transitiveDependencyVersion", "org:foo:$resolvedVersion") {
+                        if (transitiveDependencyVersion != resolvedVersion) {
+                            byAncestor()
+                        }
+                    }
                 }
             }
         }
 
         where:
-        directDependencyVersion | transitiveDependencyVersion | listVersions | resolve13
-        '[1.0,1.3]'             | '1.2'                       | true         | true
-        '1.2'                   | '[1.0,1.3]'                 | false        | false
-        '[1.0,1.2]'             | '[1.0, 1.3]'                | true         | false
-        '[1.0,1.3]'             | '[1.0,1.2]'                 | true         | true
+        directDependencyVersion | transitiveDependencyVersion | listVersions | resolvedVersion
+        '[1.0,1.3]'             | '1.2'                       | true         | '1.3' // should probably choose 1.2 instead
+        '1.2'                   | '[1.0,1.3]'                 | false        | '1.2'
+        '[1.0,1.2]'             | '[1.0, 1.3]'                | true         | '1.2'
+        '[1.0,1.3]'             | '[1.0,1.2]'                 | true         | '1.3' // should probably choose 1.2 instead
```

Comment:
Should we file a follow up issue to keep track of this? Without making a decision to change it or not now. Just something like "Reconsider resolution behavior of strictly with version ranges".

Reply:
Good idea, here it is: https://github.com/gradle/gradle/issues/10704